### PR TITLE
Block conversion: Preserve "More" order

### PR DESCRIPTION
**Pull request by mcsf**
Fixes the issue described by @swissspidy in https://github.com/WordPress/gutenberg/pull/1467#issuecomment-369926260

Kudos @swissspidy for the fix; I merely added tests.

_(drive-by pull request; can provide further details later)_

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code has proper inline documentation.

### CODE REVIEW

File: blocks/api/raw-handling/special-comment-converter.js [hunk 1]
```diff
@@ -71,3 +70,16 @@ function createMore( customText, noTeaser ) {
 	}
 	return node;
 }
+
+function replace( processedNode, newNode ) {
```
Comment (iseulde) [inline blocks/api/raw-handling/special-comment-converter.js]: Minor: Worth adding these to dom utils? We're doing these in a few places I think, even outside the paste component. Could be done separately though.

File: blocks/api/raw-handling/test/integration/classic-in.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<p>First paragraph.</p>
+<p><!--more--></p>
```
Comment (iseulde) [inline blocks/api/raw-handling/test/integration/classic-in.html]: Would it be worth adding more tests here such as `<p>First <!--more--> paragraph.</p>`?

File: blocks/api/raw-handling/test/integration/index.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ import serialize from '../../../serializer';
 
 const types = [
 	'plain',
+	'classic',
```
Comment (iseulde) [inline blocks/api/raw-handling/test/integration/index.js]: I wonder if we could consolidate the other image and iframe test into one "classic" conversion test similar to the other integration tests.

File: blocks/api/raw-handling/test/special-comment-converter.js [hunk 4]
```diff
@@ -3,6 +3,18 @@
...
 
+function trimWhitespace( string ) {
+	return string.replace( /(  +)|\n|\t/g, '' );
+}
+
+function wrapInP( string ) {
+	return `<p>${ string }</p>`;
+}
+
+function withEmptyP( string ) {
```
Comment (iseulde) [inline blocks/api/raw-handling/test/special-comment-converter.js]: I'm not sure if this is any create than without this function. `withEmptyP` leases me wondering where exactly that empty paragraph will be.

File: blocks/api/raw-handling/test/special-comment-converter.js [hunk 5]
```diff
@@ -12,38 +24,75 @@ import { deepFilterHTML } from '../utils';
...
+			<p>Third paragraph</p>`,
 			[ specialCommentConverter ]
 		);
-		// Skip the empty paragraph, which other transforms would eliminate
-		const start = output.indexOf( '</p>' ) + '</p>'.length;
 		equal(
-			output.substr( start ),
-			'<wp-block data-block="core/more" data-no-teaser=""></wp-block>'
+			trimWhitespace( output ),
+			trimWhitespace( `<p>First paragraph.</p>
```
Comment (iseulde) [inline blocks/api/raw-handling/test/special-comment-converter.js]: I don't understand why `trimWhitespace` is necessary. Shouldn't it be preserved?

Comment (iseulde) [reply blocks/api/raw-handling/test/special-comment-converter.js]: (I mean, isn't `specialCommentConverter` merely moving other nodes?)

Comment (mcsf) [reply blocks/api/raw-handling/test/special-comment-converter.js]: Yeah. As I was developing I wanted the tests to have these `input`/`output` strings neatly aligned with the rest of the code but that meant adding tab characters throughout. Not that that's wrong, but it made the tests more brittle. I'll probably remove this.

File: blocks/api/raw-handling/test/special-comment-converter.js (hunk 4 above)
Comment (mcsf) [reply blocks/api/raw-handling/test/special-comment-converter.js]: Agreed; removing simplifies things.

File: blocks/api/raw-handling/test/integration/index.js (hunk 3 above)
Comment (mcsf) [reply blocks/api/raw-handling/test/integration/index.js]: I'd like that, or something along those lines. Better left to another PR, though, no?

File: blocks/api/raw-handling/special-comment-converter.js (hunk 1 above)
Comment (mcsf) [reply blocks/api/raw-handling/special-comment-converter.js]: Done.

**Comment by mcsf** [comment]
Thanks for the review, @iseulde. Mind having one last look?

File: blocks/api/raw-handling/special-comment-converter.js (hunk 1 above)
Comment (mcsf) [reply blocks/api/raw-handling/special-comment-converter.js]: To do: actually refactor elsewhere in the code base.

**Review by aduth — APPROVED** [review]
👍

**Pull request merged**